sound/soc/sof/intel/hda-loader.c

```diff
@@ -371,6 +371,40 @@ int hda_dsp_cl_boot_firmware_iccmax(struct snd_sof_dev *sdev)
 	return ret;
 }
 
+static int hda_dsp_boot_imr(struct snd_sof_dev *sdev)
+{
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+	const struct sof_intel_dsp_desc *chip = hda->desc;
+	unsigned long mask;
+	u32 j;
+	int ret;
+
+	/* init for booting wait */
+	init_waitqueue_head(&sdev->boot_wait);
```

Comment:
this is not needed. In fact it should be removed from hda_dsp_cl_boot_firmware() as well. snd_sof_run_firmware() already does it

Reply:
> this is not needed. In fact it should be removed from hda_dsp_cl_boot_firmware() as well. snd_sof_run_firmware() already does it

What you mentioned could be true, but that should belong to another PR? Here the PR is just following what the hda_dsp_cl_boot_firmware() does.

Reply:
no I think we should remove it from here. We can remoe it from hda_dsp_cl_boot_firmware() in another PR